### DISCUSSION TITLE
Use custom icon inside CsvCatalogItem/TableStyle

### DISCUSSION THREAD

**Discussion by Oscarbralo** [TerriaJS/terriajs#7040] [Q&A] [answered]
Hello!I have a question to ask in case somebody have some knowledge about this. I am trying to use a custom icon for the `CsvCatalogItem`, inside the `marker` from the `TableStyle`. It says in the documentation that can be a 'data URI'. Does somebody knows what that means exactly?  I have tried with a .png, and a .svg icon URI, but didnt worked. The only way I have found to use this, is to load the .svg data from the icon inside the `iconset-all_maki_icons.json`. Is there any other way to use our custom png or some other kind of icons??  Thank you very much for all the help

![image (1)](https://github.com/TerriaJS/terriajs/assets/1603331/8343ab59-abfb-4e65-bab2-e4d50267edf8)

**Comment by nf-s** [comment]
Hi @Oscarbralo 

Sorry I will update the docs to be more clear. It needs to follow the [Data URL](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URLs) format - which is basically embedding a file in a URL

For example

```
data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiPz4KPHN2ZyB2ZXJzaW9uPSIxLjEiIGlkPSJzcXVhcmUtc3Ryb2tlZCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB3aWR0aD0iMTUiIGhlaWdodD0iMTUiIHZpZXdCb3g9IjAgMCAxNSAxNSI+CiAgPHBhdGggZmlsbD0icmdiKDIzMCwwLDApIiBkPSJNMTMsMTRIMmMtMC41NTIzLDAtMS0wLjQ0NzctMS0xVjJjMC0wLjU1MjMsMC40NDc3LTEsMS0xaDExYzAuNTUyMywwLDEsMC40NDc3LDEsMXYxMUMxNCwxMy41NTIzLDEzLjU1MjMsMTQsMTMsMTR6Ii8+CiAgPHBhdGggZmlsbD0iYmxhY2siIGQ9Ik0xMi43LDIuM3YxMC40SDIuM1YyLjNIMTIuNyBNMTMsMUgyQzEuNDQ3NywxLDEsMS40NDc3LDEsMnYxMWMwLDAuNTUyMywwLjQ0NzcsMSwxLDFoMTFjMC41NTIzLDAsMS0wLjQ0NzcsMS0xVjImI3hBOyYjeDk7QzE0LDEuNDQ3NywxMy41NTIzLDEsMTMsMUwxMywxeiIvPgogIAo8L3N2Zz4=
```

Will result in a red square icon like so - https://nationalmap.gov.au/#share=s-oeiMekVoP3DY5QPbtqyl0Am7L3G

**Comment by nf-s** [comment, marked as answer]
Actually, sorry there is a bug here! URLs should work, but they were being dropped.

I have a PR here that fixes the issue - https://github.com/TerriaJS/terriajs/pull/7048

It can be tested using this link - http://ci.terria.io/fix-marker-url/

Thanks for bringing it to our attention!
